### PR TITLE
[MM-47823] SendPasswordResetEmailCmdF command error handle refactoring

### PR DESCRIPTION
**Pull request by misantron**
#### Summary

SendPasswordResetEmailCmdF command error handle refactoring and tests update

#### Ticket Link

Issue: https://github.com/mattermost/mattermost-server/issues/21463
JIRA: https://mattermost.atlassian.net/browse/MM-47823

### CODE REVIEW

File: commands/user_test.go [hunk 1]
```diff
@@ -856,7 +857,23 @@ func (s *MmctlUnitTestSuite) TestSendPasswordResetEmailCmd() {
...
-	s.Run("Send one reset email and receive error", func() {
+	s.Run("Send one reset email and receive error on email validation", func() {
+		printer.Clean()
+		emailArg := "invalid.Email@example.com"
+
+		var expected error
+		expected = multierror.Append(expected, errors.Errorf("Invalid email '%s'", emailArg))
+
+		err := sendPasswordResetEmailCmdF(s.client, &cobra.Command{}, []string{emailArg})
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/user_test.go:868]: `EqualError` already checks this, no need check for `NotNil`.

File: commands/user_test.go [hunk 2]
```diff
@@ -867,8 +884,12 @@ func (s *MmctlUnitTestSuite) TestSendPasswordResetEmailCmd() {
 			Return(&model.Response{StatusCode: http.StatusBadRequest}, mockError).
 			Times(1)
 
+		var expected error
+		expected = multierror.Append(expected, errors.Wrapf(mockError, "Unable send reset password email to email %s.", emailArg))
+
 		err := sendPasswordResetEmailCmdF(s.client, &cobra.Command{}, []string{emailArg})
-		s.Require().NoError(err)
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/user_test.go:891]: Same as above

File: commands/user_test.go [hunk 3]
```diff
@@ -901,11 +927,12 @@ func (s *MmctlUnitTestSuite) TestSendPasswordResetEmailCmd() {
 		}
 
 		err := sendPasswordResetEmailCmdF(s.client, &cobra.Command{}, emailArg)
-		s.Require().NoError(err)
+		s.Require().NotNil(err)
```
Comment (noxer) [inline commands/user_test.go:930]: Same as before

File: commands/user.go [hunk 4]
```diff
@@ -498,17 +499,21 @@ func sendPasswordResetEmailCmdF(c client.Client, cmd *cobra.Command, args []stri
 		return errors.New("expected at least one argument. See help text for details")
 	}
 
+	var result *multierror.Error
+
 	for _, email := range args {
 		if !model.IsValidEmail(email) {
+			result = multierror.Append(result, errors.Errorf("Invalid email '%s'", email))
```
Comment (noxer) [inline commands/user.go:506]: Please use `fmt.Errorf` here. Also the error message should not be capitalized.

**Review by noxer — CHANGES_REQUESTED** [review]
Hi @misantron, thank you very much for your contribution. I'd like to request a few small changes. Also the linter complains about a few issues. Other than that it looks good.

File: commands/user.go [hunk 5]
```diff
@@ -498,17 +499,21 @@ func sendPasswordResetEmailCmdF(c client.Client, cmd *cobra.Command, args []stri
...
+	var result *multierror.Error
+
 	for _, email := range args {
 		if !model.IsValidEmail(email) {
+			result = multierror.Append(result, fmt.Errorf("invalid email '%s'", email))
 			printer.PrintError("Invalid email '" + email + "'")
 			continue
 		}
 		if _, err := c.SendPasswordResetEmail(email); err != nil {
+			result = multierror.Append(result, fmt.Errorf("unable send reset password email to email %s. Error: %v", email, err))
```
Comment (noxer) [inline commands/user.go:511]: You can use `%w` to wrap the error.

**Review by noxer — CHANGES_REQUESTED** [review]
Thanks for the changes, looks good. One small nitpick from me.

**Review by noxer — APPROVED** [review]
Awesome, thank you very much!

**Review by isacikgoz — APPROVED** [review]
LGTM! Thanks for the contribution and adding new test cases @misantron 🚀

**Comment by noxer** [comment]
@misantron A recent change from another PR added the import, the merge lead to duplicate imports. Could you remove those, please?

**Review by isacikgoz — APPROVED** [review]
Thanks for fixing the import issue, we need to work on the CI as this should've been flagged before merging the upstream.

**Pull request merged**